html/modules/custom/reliefweb_homepage/src/Controller/Homepage.php

```diff
@@ -270,18 +270,49 @@ public function getHeadlinesApiPayload($limit = 8) {
    * Refactor after add back redording of most read content.
    */
   public function getMostReadApiPayload($limit = 2) {
+    // Load the most-read data. This file is generated via a drush command,
+    // usually every day as the query to get the 5 most read reports is very
+    // heavy.
+    $handle = @fopen('public://most-read/most-read.csv', 'r');
+    if ($handle === FALSE) {
+      return [];
+    }
+
+    // Find the line corresponding to the entity id.
+    while (($row = fgetcsv($handle, 100)) !== FALSE) {
+      if (count($row) === 2 && $row[0] == 'front') {
+        $ids = explode(',', $row[1]);
+        break;
+      }
+    }
+
+    // Close the file.
+    if (is_resource($handle)) {
+      @fclose($handle);
+    }
+
+    // Generate the query with the most read report ids.
+    if (empty($ids)) {
+      return [];
```

Comment:
Sorry, it was not indicated but to have the same behavior as RW7:

- If there is most read data, the title should be `$this->t('Most Read <span>(last 24 hours)</span>')`.
- If there is no data from the most read CSV file, this should use the payload to get the latest updates (code before this change) and use `$this->t('Latest Updates')` as title for the block.